templates/lxc-oci.in

```diff
@@ -223,6 +223,18 @@ getcwd() {
   return
 }
 
+is_same_mountpoint() {
+  local path1="$1"
+  local path2="$2"
+  local mount1
+  local mount2
+
+  # make sure bind mounts are covered
+  mount1=$(df --output=target "$path1" | tail -n 1)
+  mount2=$(df --output=target "$path2" | tail -n 1)
+  [ "$mount1" = "$mount2" ]
+}
```

Comment:
This looks very hackish, shouldn't you be using `stat -f -c '%i' PATH` instead?

Reply:
I agree it's hackish. Unfortunately `stat -f -c '%i' PATH` won't spot bind mounts if both dirs have the same filesystem.

```
root@node1:~# touch /var/lib/lxc/x ; ln /var/lib/lxc/x /var/cache/lxc/x
ln: failed to create hard link '/var/cache/lxc/x' => '/var/lib/lxc/x': Invalid cross-device link
root@node1:~# stat -f -c '%i' /var/lib/lxc/ /var/cache/lxc/
1ac394251fdb6098
1ac394251fdb6098
root@node1:~# df --output=target /var/lib/lxc/ | tail -n 1
/var/lib/lxc
root@node1:~# df --output=target /var/cache/lxc/ | tail -n 1
/
```